src/GitHub.App/Factories/HostCacheFactory.cs

```diff
@@ -39,5 +39,23 @@ public IBlobCache Create(HostAddress hostAddress)
 
         IOperatingSystem OperatingSystem { get { return operatingSystem.Value; } }
         IBlobCacheFactory BlobCacheFactory { get { return blobCacheFactory.Value; } }
+
+        bool disposed;
+        protected virtual void Dispose(bool disposing)
```

Comment:
Yeah, it's a bit of a waste of lines of code, but code analysis rules will always complain if this pattern isn't followed for non-sealed classes, and given that this is an annoyingly hard pattern to get right, I'm taking the approach of copy-pasting the pattern across all our implementations to make sure it's always correct. I've had more problems with random copy-pasting particular implementations of Dispose that were only appropriate for some specific scenario, which then fail subtly, than I've had just using the more verbose pattern everywhere, so that's what I'm doing here 😄 

Since the blob cache factory now needs to be disposed, we should actually be using `ExportLifetimeProvider` instead of `Lazy`, which is the proper way of controlling MEF disposals and object lifetime. I might throw you another PR with that.